lib/kernel-manager.coffee

```diff
@@ -11,6 +11,7 @@ module.exports =
 class KernelManager
     constructor: ->
         @_runningKernels = {}
+        @_kernelSpecs = []
```

Comment:
On 11/07/16 10:55, Lukas Geiger wrote:
>>    @n-riesco <https://github.com/n-riesco> I think @_kernelSpecs should be initialised with the user settings; i.e. @_kernelSpecs = @parseKernelSpecSettings(). Is there a case when we wouldn't this to happen?
> 
> The parsing of user settings should happen in |updateKernelSpecs|, otherwise |hydrogen:update-kernels| wouldn't help if the settings are changed by the user.

One thing doesn't preclude the other, but it makes me think that `@updateKernelSpecs` could be split into two methods `@getKernelSpecsFromSettings` and `@getKernelSpecsFromJupyter`.

The constructor would call `@getKernelSpecsFromSettings`.
And `@updateKernelSpecs` woudl call `@getKernelSpecsFromSettings` and `@getKernelSpecsFromJupyter`.